src/modules/app/StateType.js

```diff
@@ -37,6 +39,11 @@ export type FileCacheStateType = {
   |}
 }
 
+export type ErrorStateType = {|
+  +error: true,
+  +message: string
```

Comment:
You can change this to: 
```
errorMessage: ?string
```

Reply:
or just `errorMessage` because if it is null then there should be a full `CategoryRouteStateType`

Reply:
Nope, not possible. See https://flow.org/en/docs/types/unions/#toc-disjoint-unions-with-exact-types . A full `CategoryRouteStateType` AND an error makes no sense. So if there is no `CategoryRouteStateType`, there has to be an `ErrorStateType` and vice versa. This should really be an XOR

Reply:
Yes this is a  "either/or" relationship. But this structure contains redundancy if I get it right.
Like: If message is not null or undefined then error is always true.

Reply:
Yeah I know, but this is the recommended way of flow to do smth like this. If this redundancy is removed,  flow throws errors, because flow doesn't get it anymore. E.g. if you have types `type ErrorType = {|
  error:  string
|}` and `type SuccessType = {|
   data: XY,
  moreData: UV
|}` and `type SomeType = ErrorType | SuccessType` and you check for error like this:
`if (someObject.error) {
  do something
} else {
  do something with someObject.data
}`,
flow throws an error that someObject.data is not defined. You'd always have to check for all props you want to use,  e.g. `if (someObject.data && someObject.moreData) { do something with data and moreData }`

Reply:
Hm if I understand you correctly something like this should not work, right?

```
type AaType = {|
  errorMessage: string
|}

type BbType = {|
  result: any
|}

type CcType = AaType | BbType

const c: CcType = {result: 'true'}

if (c.errorMessage) {
  console.log(c.errorMessage)
  const r: AaType = c
  console.log(r)
}

if (c.result) {
  console.log(c.result)
  const r: BbType = c
  console.log(r)
}
```

Reply:
Okay, we have to check for `undefined` to make it working.